### ISSUE TITLE
dedicated server issues

### ISSUE THREAD

**Issue by Jalewis1990** [zolantris/ValheimRaft#12]
on my server raft falls apart when returning to it. old rafts from previous versions that I could not interact with after crafting still remain ghost rafts.  

 Mod List:

   -Valheim + (by grantapher)
   -hookgenpatcher
   -jotunn the valheim library
   -seed totem
   -plant everything
   -weaponadditions
   -magicplugin
   -multi user chest
   -spyglass
   -odinsfoodbarrels
   -rune magic
   -speedy paths
   -balronds shipyard

i have not replicated in isolation

**Comment by zolantris** [comment]
I'll update the readme. It's replicable. And I did not test on server 😞

**Comment by zolantris** [comment]
Looks like the follow is happening
- the mod on the server attempts to load. And fails (or does not match the client copy created content).
- It then garbage collects the resources that broke. This happens on load. 
- On the client side, creating things will work (sometimes) but the server will consistently interfere. Then on the user log off or save or server startup it will garbage collect the references that it deems broken with valheim raft.

**Comment by Jalewis1990** [comment]
If you need any help testing on a server I would be more than happy to get you on mine

**Comment by zolantris** [comment]
Edit: 1.6.1 does not fix this. But I have figured out part of the issue. Somehow the game ZDOs are not set as permanent and therefore are garbage collected by the server raft mod. The Server also can garbage collect if it has a mod that does this already.

Looks like the fix is finding out why the ZDOs are not considered permanent even though there is a bunch of code setting the as such.

**Comment by jackerzhaques** [comment]
Not sure if this helps or if you're still working on this issue. I notice that it only happens rarely on my server, and only when using portals. If I teleport to the boat, sometimes it will get stuck in an infinite loading screen. After I force quit on client side and rejoin the server, the raft is missing. The only thing left is a base raft with no floors. I've never had the issue if I don't use teleporters.

**Comment by zolantris** [comment]
Yeah portals are unstable. I'm looking into it. Probably have to re-write the rendering logic to directly hook into zone loading (especially for multiple boats).

Other users have reported 2 teleports consecutively back and forth will cause the problem.

**Comment by LtDoc** [comment]
We're running a dedi with about 10 people and also have pretty severe issues that crop up where ships will be fine for awhile but suddenly a player will log on and half the ship is gone. I've tried recreating steps for reporting but it's mostly random on when it happens. Portals don't cause it as as far as players rendering and unrendering it go, it mostly appears to happen between server restarts.

**Comment by zolantris** [comment]
Are players still logged in but moving when the server restarts? Also is the desync happening when multiple players are on the same boat?
